ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/RegisterSchemaCallback.java

```diff
@@ -43,17 +44,66 @@ public void onDeserializationFailure(
   ) {
     final String sourceSubject = source + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX;
     final String changelogSubject = changelog + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX;
+
+    // all schema registry events start with a magic byte 0x0 and then four bytes
+    // indicating the schema id - we extract that schema id from the data that failed
+    // to deserialize and then register it into the changelog subject
+    final int id = ByteBuffer.wrap(data, 1, Integer.BYTES).getInt();
+    final SchemaRegisterEvent event = new SchemaRegisterEvent(id, sourceSubject, changelogSubject);
+
     try {
-      // all schema registry events start with a magic byte 0x0 and then four bytes
-      // indicating the schema id - we extract that schema id from the data that failed
-      // to deserialize and then register it into the changelog subject
-      final int id = ByteBuffer.wrap(data, 1, Integer.BYTES).getInt();
-
-      LOG.info("Trying to fetch & register schema id {} under subject {}", id, changelogSubject);
-      final ParsedSchema schema = srClient.getSchemaBySubjectAndId(sourceSubject, id);
-      srClient.register(changelogSubject, schema);
-    } catch (IOException | RestClientException e) {
-      LOG.warn("Failed during deserialization callback for topic " + source, e);
+      if (!failedAttempts.contains(event)) {
+        LOG.info("Trying to fetch & register schema id {} under subject {}", id, changelogSubject);
+        final ParsedSchema schema = srClient.getSchemaBySubjectAndId(sourceSubject, id);
+        srClient.register(changelogSubject, schema);
+      }
+    } catch (Exception e) {
+      LOG.warn("Failed during deserialization callback for topic {}. "
```

Comment:
Does it make sense to use some kind of retry n times, with a backoff, or is this likely to work after an additional attempt?

Reply:
see inline responses to @vcrfxia's review